### ISSUE TITLE
add units to target catalog table

### ISSUE THREAD

**Issue by moustakas** [desihub/desitarget#356]
None of the columns in the output targeting catalog have units.  Do we care?
```python
import fitsio
from astropy.table import Table
tt = Table(fitsio.FITS('/global/projecta/projectdirs/desi/target/catalogs/dr7.1/0.23.0/targets-dr7.1-0.23.0.fits')[1].read(rows=0))
for col in tt.colnames:
    if tt[col].unit is not None:
        print('Column {} has units!'.format(col))
```

**Comment by sbailey** [comment]
Ideally we would include units in the fits files themselves, as well as in the data model.  Low priority.

**Comment by geordie666** [comment]
Addressed in #638.
